### PR TITLE
Properly store User Subclass in Storage

### PR DESCRIPTION
**Pull request by dplewis**
@taivo Let me know if any changes can be made to replicate your issue

### CODE REVIEW

**Comment by dplewis** [comment]
@taivo The issue here is a fundamental one.

CustomUser -> extends Parse.User -> super('attributes') -> Parse.User extends Parse.Object -> super('_User') <- CustomUser className never gets set.

**Comment by dplewis** [comment]
@acinader This is the easiest fix I can come up with.

The issue here is that subclass users don't automatically have a className (must be explicitly set or passed in [here](https://github.com/parse-community/Parse-SDK-JS/blob/master/src/ParseUser.js#L46) somehow). The second issue is that even if a className exist, it's hardcoded before it saves to disk.

File: src/ParseUser.js [hunk 1]
```diff
@@ -869,7 +869,7 @@ const DefaultController = {
   updateUserOnDisk(user) {
     const path = Storage.generatePath(CURRENT_USER_KEY);
     const json = user.toJSON();
-    json.className = '_User';
+    json.className = user.className || '_User';
```
Comment (acinader) [inline src/ParseUser.js]: how about 

`json.className = this.constructor.name === 'ParseUser' ? '_User' : this.constructor.name;`

i think that'll work and then you don't have to worry about setting this.className.

Comment (dplewis) [reply src/ParseUser.js]: Perfect!

Comment (dplewis) [reply src/ParseUser.js]: Looks like we have a flaky test. I can't restart the build the button is missing, can you?

**Comment by taivo** [comment]
This is good. When you guys have it merged, I can switch my use case to this and test for a few days.

**Pull request merged**